From c0dab503d8261781b5c88c601c645330315ead08 Mon Sep 17 00:00:00 2001
From: Adnan Begovic <adnan@cyngn.com>
Date: Wed, 30 Sep 2015 12:17:41 -0700
Subject: [PATCH 10/10] build: Make sure we're building our secondary resource
 package as dependency.

  This allows us to reference internal or public resources without
  going explicitily through the platform sdk.

Change-Id: Iba361a3dcd20bc62a06e600e0240fb09d4e7bda4
---
 core/package_internal.mk | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/core/package_internal.mk b/core/package_internal.mk
index 80d386d4b..ad600b06b 100644
--- a/core/package_internal.mk
+++ b/core/package_internal.mk
@@ -493,6 +493,18 @@ framework_res_package_export := \
 else # LOCAL_SDK_RES_VERSION
 framework_res_package_export := \
     $(call intermediates-dir-for,APPS,framework-res,,COMMON)/package-export.apk
+
+# Avoid possible circular dependency with our platform-res
+ifneq ($(LOCAL_IGNORE_SUBDIR), true)
+lineage_plat_res_package_export := \
+    $(call intermediates-dir-for,APPS,org.lineageos.platform-res,,COMMON)/package-export.apk
+endif # LOCAL_IGNORE_SUBDIR
+
+ifneq ($(LOCAL_IGNORE_SUBDIR), true)
+lineage_plat_res_package_export_deps := \
+    $(dir $(lineage_plat_res_package_export))src/R.stamp
+endif # LOCAL_IGNORE_SUBDIR
+
 endif # LOCAL_SDK_RES_VERSION
 endif # LOCAL_NO_STANDARD_LIBRARIES
 
@@ -505,6 +517,14 @@ all_library_res_package_export_deps := \
     $(framework_res_package_export) \
     $(foreach lib,$(LOCAL_RES_LIBRARIES),\
         $(call intermediates-dir-for,APPS,$(lib),,COMMON)/src/R.stamp)
+
+ifneq ($(LOCAL_IGNORE_SUBDIR), true)
+all_library_res_package_exports += \
+    $(lineage_plat_res_package_export)
+all_library_res_package_export_deps += \
+    $(lineage_plat_res_package_export_deps)
+endif # LOCAL_IGNORE_SUBDIR
+
 $(resource_export_package) $(R_file_stamp) $(LOCAL_BUILT_MODULE): $(all_library_res_package_export_deps)
 $(LOCAL_INTERMEDIATE_TARGETS): \
     PRIVATE_AAPT_INCLUDES := $(all_library_res_package_exports)
-- 
2.17.1

